### PR TITLE
[dfs] fixups: page cache

### PR DESCRIPTION
**Pull request by polarvid**
## 拉取/合并请求描述：(PR description)

[

#### 为什么提交这份PR (why to submit this PR)

This patch includes following fixups:

- missing unmap of pages when they are recycled
- while instruction page of user is remapped, the neccessary synchronization of data cache to instruction cache is required for machines implementing Harvard Architecture.

#### 在什么测试环境下测试通过 (what is the test environment)

]

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 代码是高质量的 Code in this PR is of high quality
- [x] 已经使用[formatting](https://github.com/mysterywolf/formatting) 等源码格式化工具确保格式符合[RT-Thread代码规范](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_en.md)

### CODE REVIEW

File: components/dfs/dfs_v2/src/dfs_file_mmap.c [hunk 1]
```diff
@@ -272,13 +272,14 @@ rt_err_t on_varea_shrink(struct rt_varea *varea, void *new_vaddr, rt_size_t size
...
     }
-    else if (varea_start < (char *)new_vaddr)
+    else
     {
         rm_start = varea_start;
         rm_end = new_vaddr;
     }
 
-    return unmap_pages(varea, rm_start, rm_end);
+    unmap_pages(varea, rm_start, rm_end);
```
Comment (geniusgogo) [inline components/dfs/dfs_v2/src/dfs_file_mmap.c:281]: 拆成两句，不加返回值判断？

File: components/dfs/dfs_v2/src/dfs_file_mmap.c [hunk 2]
```diff
@@ -308,9 +309,9 @@ rt_err_t on_varea_split(struct rt_varea *existed, void *unmap_start, rt_size_t u
         }
 
         unmap_pages(existed, unmap_start, (char *)unmap_start + unmap_len);
+        unmap_pages(existed, subset->start, (char *)subset->start + subset->size);
```
Comment (geniusgogo) [inline components/dfs/dfs_v2/src/dfs_file_mmap.c:312]: 会映射失败么？需要检查返回值吗？

File: components/dfs/dfs_v2/src/dfs_pcache.c [hunk 3]
```diff
@@ -1272,6 +1275,7 @@ void *dfs_aspace_mmap(struct dfs_file *file, struct rt_varea *varea, void *vaddr
             {
                 ret = page->page;
                 map->varea = varea;
+                rt_hw_cpu_icache_invalidate(vaddr, ARCH_PAGE_SIZE);
```
Comment (geniusgogo) [inline components/dfs/dfs_v2/src/dfs_pcache.c:1278]: 感觉突然在这个地方刷cache有点莫名其妙呢？第一现场是啥？

Comment (polarvid) [reply components/dfs/dfs_v2/src/dfs_pcache.c]: 这里重新建立一段映射就是需要调用 cache API 清除重名缓存的根本原因。如果放到 unmap API 里面，因为要判断的条件非常多，而且此场景发生的概率不高，增加大量条件判断会影响效率。综合判断，在此处维护 icache 同步就是最合理的。

**Review by geniusgogo — APPROVED** [review]

**Review by Guozhanxin — APPROVED** [review]

**Pull request merged**